pkg/sql/importer/import_processor.go

```diff
@@ -178,11 +185,26 @@ func (idp *readImportDataProcessor) Start(ctx context.Context) {
 	ctx = idp.StartInternal(ctx, readImportDataProcessorName)
 	// We don't have to worry about this go routine leaking because next we loop over progCh
 	// which is closed only after the go routine returns.
-	go func() {
+	idp.wg = ctxgroup.WithContext(ctx)
+	idp.wg.GoCtx(func(ctx context.Context) error {
 		defer close(idp.progCh)
 		idp.summary, idp.importErr = runImport(ctx, idp.flowCtx, &idp.spec, idp.progCh,
 			idp.seqChunkProvider)
-	}()
+		return nil
+	})
+}
+
+func (idp *readImportDataProcessor) ConsumerClosed() {
+	idp.close()
+}
+
+func (idp *readImportDataProcessor) close() {
+	if idp.Closed {
+		return
```

Comment:
one question i have for the queries team: does `idp.Closed==true` imply that the context passed to Start() has been closed? I couldn't understand this easily from poking around processorBase.